src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Linux/Network/LinuxNetworkUtilizationParser.cs

```diff
@@ -0,0 +1,154 @@
+
+    /// <remarks>
+    /// Reads the contents of a file located at _tcp4 and parses it to extract information about the TCP/IP state info on the system.
+    /// </remarks>
+    public TcpStateInfo GetTcpIPv4StateInfo() => GetTcpStateInfo(_tcp);
+
+    /// <remarks>
+    /// Reads the contents of a file located at _tcp6 and parses it to extract information about the TCP/IP state info on the system.
+    /// </remarks>
+    public TcpStateInfo GetTcpIPv6StateInfo() => GetTcpStateInfo(_tcp6);
+
+    public LinuxNetworkUtilizationParser(IFileSystem fileSystem)
+    {
+        _fileSystem = fileSystem;
+    }
+
+    /// <remarks>
+    /// The method is used to read Span data and calculate the TCP state info.
+    /// Refer <see href="https://www.kernel.org/doc/Documentation/networking/proc_net_tcp.txt">proc net tcp</see>.
+    /// </remarks>
+    private static void UpdateTcpStateInfo(ReadOnlySpan<char> buffer, TcpStateInfo tcpStateInfo)
+    {
+        const int Base16 = 16;
+        ReadOnlySpan<char> line = buffer.TrimStart();
+
+#if NET8_0_OR_GREATER
+        const int Target = 5;
+        Span<Range> range = stackalloc Range[Target];
+
+        // on .NET 8+, if capacity of destination range array is less than number of ranges found by ReadOnlySpan<T>.Split(),
+        // the last range in the array will get all the remaining elements of the ReadOnlySpan.
+        // therefore we request 5 ranges instead of 4, and then range[Target - 2] will have the range we need without the remaining elements.
+        int numRanges = line.Split(range, ' ', StringSplitOptions.RemoveEmptyEntries);
+#else
+        const int Target = 4;
+        Span<StringRange> range = stackalloc StringRange[Target];
+
+        // in our StringRange API, if capacity of destination range array is less than number of ranges found by ReadOnlySpan<T>.TrySplit(),
+        // the last range in the array will get the last range as expected, and all remaining elements will be ignored.
+        // hence range[Target - 1] will have the last range as we need.
+        _ = line.TrySplit(" ", range, out int numRanges, StringComparison.OrdinalIgnoreCase, StringSplitOptions.RemoveEmptyEntries);
+#endif
+        if (numRanges < Target)
+        {
+            Throw.InvalidOperationException($"Could not split contents. We expected every line to contain more than {Target - 1} elements, but it has only {numRanges} elements.");
+        }
+
+#if NET8_0_OR_GREATER
+        ReadOnlySpan<char> tcpConnectionState = line.Slice(range[Target - 2].Start.Value, range[Target - 2].End.Value - range[Target - 2].Start.Value);
+#else
+        ReadOnlySpan<char> tcpConnectionState = line.Slice(range[Target - 1].Index, range[Target - 1].Count);
+#endif
+
+        // until this API proposal is implemented https://github.com/dotnet/runtime/issues/61397
+        // we have to allocate & throw away memory using ToString():
+        switch ((LinuxTcpState)Convert.ToInt32(tcpConnectionState.ToString(), Base16))
```

Comment:
🤔  What does `tcpConnectionState.ToString()` produce? 
It'd be great to add a comment here.

Reply:
What do you mean exactly? 
.ToString(), when called on the ReadOnlySpan<shar>, produces a new string.

Reply:
I was more curious what the content of the variable - i.e., what's the string it contains.

Reply:
Update the comment